standalone/alertAndPluginDetails.js

```diff
@@ -0,0 +1,40 @@
+extAlert = org.parosproxy.paros.control.Control.getSingleton().
+    getExtensionLoader().getExtension(
+        org.zaproxy.zap.extension.alert.ExtensionAlert.NAME) 
+
+extAscan = org.parosproxy.paros.control.Control.getSingleton().
+    getExtensionLoader().getExtension(
+        org.zaproxy.zap.extension.ascan.ExtensionActiveScan.NAME);
+
+extPscan = org.parosproxy.paros.control.Control.getSingleton().
+    getExtensionLoader().getExtension(
+        org.zaproxy.zap.extension.pscan.ExtensionPassiveScan.NAME);
+
+var pf = Java.type("org.parosproxy.paros.core.scanner.PluginFactory");
+
+printHeaders();
+
+if (extAlert != null) {
+    var Alert = org.parosproxy.paros.core.scanner.Alert;
+    var alerts = extAlert.getAllAlerts();
+    for (var i = 0; i < alerts.length; i++) {
+        var alert = alerts[i]
+        printAlert(alert);
+    }
+}
+
+function printHeaders() {
+    print('AlertName\tSource:PluginName\tWASC\tCWE');
+}
+
+function printAlert(alert) {
+    var scanner = '';
+    if (alert.getSource() == Alert.Source.ACTIVE) {
+        scanner = pf.getLoadedPlugin(alert.getPluginId()).getName();

```

Comment:
It should check that the scanner exists before getting the name (same for passive), it might happen that the corresponding add-on is not installed.
```
TypeError: null has no such function "getName"
<eval>:33 TypeError: null has no such function "getName"
```